### PR TITLE
Dialog2.0

### LINKED ISSUE #1249: 2.0 - Update Dialog
This is a tracking issue for updating Dialog

**Scope:** Dialog

### PR DESCRIPTION
**Pull request by prateekbh**
Fixes #1249
Confirms to #1293 
- Adds `onClose` prop.
- Adds `onOpen` prop.
- Fixes Exports.
- Removes dependency on `@bind` because with arrow functions `module` build needs no transpilation at all. Where as with `@bind` it adds some code which will be in module build as well

### CODE REVIEW

File: packages/dialog/src/action.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  protected componentName = 'dialog__button';
+  protected mdcProps = [];
+
+  protected materialDom(allProps) {
+    const {accept, cancel, children, props} = allProps;
+    let actionString = 'false';
+    if (accept) {
+      actionString = 'accept';
+    } else if (cancel) {
+      actionString = 'close';
```
Comment (cromefire) [inline packages/dialog/src/action.tsx]: Just to be sure, this are the only options right?

Comment (prateekbh) [reply packages/dialog/src/action.tsx]: They can be anything but these are the default ones

Comment (cromefire) [reply packages/dialog/src/action.tsx]: Shouldn't the user have a choice then?

Comment (prateekbh) [reply packages/dialog/src/action.tsx]: They have a choice by adding a prop, but I guess we should create a`action` prep to make it easier?

Comment (cromefire) [reply packages/dialog/src/action.tsx]: Yes and then just replace `false` with
```typescript
action ? action : 'false' 
```

Comment (cromefire) [reply packages/dialog/src/action.tsx]: (and add it to the docs)

**Pull request merged**